src/wallet/wallet.cpp

```diff
@@ -4057,16 +4138,17 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
         // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
         walletInstance->SetMinVersion(FEATURE_LATEST);
 
-        if ((wallet_creation_flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
-            //selective allow to set flags
-            walletInstance->SetWalletFlag(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
-        } else if (wallet_creation_flags & WALLET_FLAG_BLANK_WALLET) {
-            walletInstance->SetWalletFlag(WALLET_FLAG_BLANK_WALLET);
-        } else {
+        if ((wallet_creation_flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS) &&
```

Comment:
@jnewbery Existing code would set at most one of the two flags (disable privkeys or blank_wallet), and would not set any other wallet flags. As I am introducing more wallet flags, I needed those to be set, but because the existing code will *ignore* the blank_wallet flag if the *disable privkeys* flag is set, I had to do some tweakery to not modify the existing behavior.

If everyone agrees that suddenly setting both disable_privkeys AND blank_wallet is harmless, this code can be simplified down to just doing `->SetWalletFlags(...)` without the if block, but I didn't feel confident making that decision myself.

Reply:
Ah. I see now. Yes, this makes sense, although it's a little confusing the way that the generic `setWalletFlags()` is called between two blocks of blank wallet specific code.

I've done some grepping and it seems to me that it's fine for `WALLET_FLAG_BLANK_WALLET` and `WALLET_FLAG_DISABLE_PRIVATE_KEYS` to be set on the same wallet. @achow101 should know for sure.

Reply:
It's fine to set both `WALLET_FLAG_BLANK_WALLET` and `WALLET_FLAG_DISABLE_PRIVATE_KEYS` on a wallet, though redundant. `WALLET_FLAG_BLANK_WALLET` will be unset whenever something is imported.

Reply:
I wish the `BLANK_WALLET` flag was called something more like `WALLET_IS_ALREADY_INITIALIZED` or `WALLET_DO_NOT_REINITIALIZE` since the only effect of the flag is to skip the `fFirstRun` code path when the wallet is loaded. I think the term "blank wallet" is good as a createwallet option name, but less good as a flag name.

Reply:
Removed the redundant if block.